examples/ws2812b_bitbang/ws2812bbitbang_driver.h

```diff
@@ -0,0 +1,65 @@
+// Include necessary libraries
+#include <stdlib.h>
+
+// Initialise the variables for animation
+uint8_t flag = 0;
+uint32_t count = 1;
+
+// Initialise the variable for the srand seed
+uint16_t ADC_val;
+
+// Define an array to buffer the color data for each LED in the strip
+uint8_t BUFFER_LEDS[num_leds][3] = {};
+
+// Send a single bit using the GD pinwiggle protocol.
+// Check mark/space ratio of the data on GPIO D6 with an oscilloscope (or Logic analyser?).
+void LED_SendBit(uint8_t bit)
+{
+    if (bit) {
+    //// Send a 1 bit
+        GPIOD->BSHR = 1 << 6; // put pin C4 high and wait for 800nS
+            __asm__("nop");__asm__("nop");__asm__("nop");__asm__("nop");
+            __asm__("nop");__asm__("nop");__asm__("nop");__asm__("nop");
+            __asm__("nop");__asm__("nop");__asm__("nop");__asm__("nop");

```

Comment:
My guess it is because it needs to be exactly 800nS and a for loop would be more difficult to exactly time.

I did similar NOP galore when doing timing critical stuff similar to WS2812.

Note that it will depend on the clock speed, so some precompiler magic is usually required to adjust for a various clock speeds.